spacksites/spack-env-templates/dev1/spack.yaml

```diff
@@ -0,0 +1,41 @@
+spack:
+  specs:
+  - matrix:
+    - - 'bcftools@1.16'
+      - 'beast2@2.7.4'
+      - 'bedtools2@2.31.0'
+      - 'bwa@0.7.17'
+      - 'gatk@4.4.0.0'
+      - 'gzip@1.13'
+      - 'hdf5@1.14.3 +cxx +fortran -mpi +hl'
+      - 'hdf5@1.14.3 +fortran +mpi +hl'
+      - 'htslib@1.17'
+      - 'openmpi@4.1.6 fabrics=psm2,cma,ucx,knem,xpmem,ofi schedulers=sge'
+      # - 'orca@5.0.3'
+      - 'python@3.11.6'
+      - 'netcdf-c@4.9.2 -mpi'
+      - 'netcdf-fortran@4.6.1'
+      - 'picard@2.26.2'
+      - 'samtools@1.17'
+      - 'vcftools@0.1.16'
+    - - '%gcc@12.3.0'
+  # CUDA packages, compile them with gcc@12
+  - matrix:
+    - - 'cp2k@2023.2 +cuda +mpi +openmp ~sirius cuda_arch=80 build_system=cmake ^dbcsr cuda_arch=80 ^libxsmm@1.17'
+      - 'cuda@12.2.0' # v12.2.0 matches the driver installed on Myriad
+      # - 'namd@2.14 +cuda cuda_arch=80'
+      # - 'py-alphafold@2.2.4 +cuda cuda_arch=80 ^py-scipy@1.11.4 ^py-pythran@0.12.2 ^openmm@7.5.1 +cuda cuda_arch=80'
+      # - 'vasp@6.3.2 +cuda'
+    - - '%gcc@12.3.0'
+  concretizer:
+    unify: when_possible
+  view: false
+  modules:
+    default:
+      tcl:
+        projections:
+          all:  '{name}/{version}/{compiler.name}-{compiler.version}'
+          ^mpi: '{name}/{version}-{^mpi.name}/{compiler.name}-{compiler.version}'
```

Comment:
This is to set the name of the modules when MPI is available.  I tried to follow the naming scheme I saw currently on Myriad, but I'm not married to this solution, I'm happy to discuss improvements.  We might want to include Cuda, too, for example, the only caveat is that we'd need to consider also all cross-cases (only mpi, only cuda, mpi+cuda), unfortunately the system doesn't seem to be very composable.